keel-scheduler/src/main/kotlin/com/netflix/spinnaker/keel/scheduler/handler/ConvergeIntentHandler.kt

```diff
@@ -60,7 +61,7 @@ class ConvergeIntentHandler
 
     if (clock.millis() > message.timeoutTtl) {
       log.warn("Intent timed out, canceling converge for {}", value("intent", message.intent.id))
-      applicationEventPublisher.publishEvent(ConvergenceTimeoutEvent(message))
+      applicationEventPublisher.publishEvent(IntentConvergeTimeoutEvent(message.intent))
```

Comment:
What's the reason that all these `.publishEvent`s happen before the action is taken instead of after?